smart_campus/app/views.py

```diff
@@ -1,3 +1,149 @@
 from django.shortcuts import render
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from django.contrib.auth import authenticate, login, logout
+from django.http import HttpResponseRedirect, JsonResponse
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
 
-# Create your views here.
+from .models import (
+    User, Reward, Permission,
+    Station,
+)
+
+
+@csrf_exempt
+@require_POST
+def signup(request):
+    """Signup API for APP users
+
+    Register new users
+
+    """
+    email = request.POST.get('email')
+    password = request.POST.get('password')
+    nickname = request.POST.get('nickname')
+
+    if User.objects.filter(email=email).exists():
+        return JsonResponse({'status': 'false', 'message': 'User already exists!'})
+
+    try:
+        User.objects.create_user(email, password, nickname)
+    except ValueError as error:
+        return JsonResponse({'status': 'false', 'message': error})
+
+    return JsonResponse({'status': 'true', 'message': 'Success'})
+
+
+@csrf_exempt
+@require_POST
+def signin(request):
```

Comment:
It seems that you use `signin`, `signout` as API and `log_in`, `log_out` as pages and it is kind of confusing.
One of the solutions might be adding a `_page` suffix to the views that render HTML pages.

Reply:
Ok!